xbmc/addons/PluginSource.cpp

```diff
@@ -102,7 +102,8 @@ bool CPluginSource::IsType(TYPE type) const
        || (type == ADDON_AUDIO && Provides(AUDIO))
        || (type == ADDON_IMAGE && Provides(IMAGE))
        || (type == ADDON_GAME && Provides(GAME))
-       || (type == ADDON_EXECUTABLE && Provides(EXECUTABLE)));
+       || (type == ADDON_EXECUTABLE && Provides(EXECUTABLE))
+       || CAddon::IsType(type)); // Fallback also to other types e.g. ADDON_SCRIPT is asked
```

Comment:
Why is this change needed?

Reply:
It doesn't make much sense to me. A plugin is a plugin, why do you check it against `ADDON_SCRIPT` ?

Reply:
The directory ask always for `ADDON_SCRIPT` but the scripts are also handled by `CPluginSource` where primary for `ADDON_PLUGIN`, without it becomes never a true if asked for.